### PR TITLE
Add vote for 2022/06/14: Merge-ready protocol service pack (final round) + common tests

### PR DESCRIPTION
**Pull request by arwer13**
### Voting and tests for merge-ready protocol upgrade (final round)

1. Revoke DEPOSIT_ROLE from old DepositSecurityModule [`0xDb149235B6F40dC08810AA69869783Be101790e7`](https://etherscan.io/address/0xDb149235B6F40dC08810AA69869783Be101790e7)
2. Grant DEPOSIT_ROLE to new DepositSecurityModule [`0x710B3303fB508a84F10793c1106e32bE873C24cd`](https://etherscan.io/address/0x710B3303fB508a84F10793c1106e32bE873C24cd`)
3. Set lastDepositBlock of DepositSecurityModule to block number 14985614

UPD (14-Jun-2022): Previously announced vote item 4 has been removed.

#### Context
- [Announcement on the Lido research forum](https://research.lido.fi/t/announcement-merge-ready-protocol-service-pack/2184)
- [Snapshot vote](https://snapshot.org/#/lido-snapshot.eth/proposal/0x37f68adc95bf8adb19dc14c0f0734bc4f6330274d3e63fbafae0a12373e8e997)
- [Twitter announcements](https://twitter.com/LidoFinance/status/1526121726639316994) :bird: 
- [Contracts](https://github.com/lidofinance/lido-dao/releases/tag/v1.2.0-rc.1)

### CODE REVIEW

**Comment by TheDZhon** [comment]
Looks good to deploy and test on Görli testnet :robot:

File: tests/common/regression/test_regression_burn_shares.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+def stranger(accounts):
+    return accounts[0]
+
+
+@pytest.fixture(scope="module")
+def dao_voting_as_eoa(accounts, dao_voting):
+    return accounts.at(dao_voting.address, force=True)
+
+
+@pytest.fixture(scope="module", autouse=(start_and_execute_votes is not None))
```
Comment (folkyatina) [inline tests/common/regression/test_regression_burn_shares.py:23]: Had a hard times understanding what the actual condition for executing the fixture here. `is not None` check is lacking answers on a Why question.

Comment (folkyatina) [reply tests/common/regression/test_regression_burn_shares.py:23]: having a variable like `is_there_any_vote_scripts = start_and_execute_votes is not None` will help

Comment (folkyatina) [reply tests/common/regression/test_regression_burn_shares.py:23]: Or we can return a tuple from the `get_start_and_execute_votes_func()`

File: tests/common/tx_tracing_helpers.py [hunk 2]
```diff
@@ -28,6 +29,8 @@ def display_voting_call_trace(tx: TransactionReceipt) -> None:
 
 
 def count_vote_items_by_events(tx: TransactionReceipt, voting_addr: str) -> int:
+    voting_fetched_instance_unused = Contract.from_explorer(voting_addr)
```
Comment (folkyatina) [inline tests/common/tx_tracing_helpers.py:32]: Add a comment why we need this magic call here?

File: utils/import_current_vote.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+def get_vote_script_files() -> List[str]:
+    """Return List of abs paths to vote scripts"""
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    dir_path = os.path.join(os.path.split(dir_path)[0], 'scripts')
+    vote_files = glob.glob(os.path.join(dir_path, 'vote_*.py'))
+    return vote_files
+
+
+def get_start_and_execute_votes_func() -> Optional[Callable]:
+    vote_files = get_vote_script_files()
```
Comment (folkyatina) [inline utils/import_current_vote.py:19]: Sorry, I didn't get, why we need a closure here? Can we make it two plain stupid functions instead?

**Review by folkyatina — APPROVED** [review]
Actually, minor notes and some ideas to improve a testing framework. Nothing critical to hinder the merge.

File: README.md [hunk 4]
```diff
@@ -148,6 +150,51 @@ To mask obsoleted tests and prevent them from running by `brownie test` even
 when residing in archive directory, please consider to rename them:
 `test_` → `xtest_`.
 
+## Common tests
+
+There are two groups of common tests in `tests` directory: regression
+(`tests/test_regression*.py`) and snapshot (`tests/test_snapshot*.py`).
```
[the remark this answers is not in the document]
Comment (TheDZhon) [reply README.md:156]: yep, nice one

File: README.md [hunk 5]
```diff
@@ -148,6 +150,51 @@ To mask obsoleted tests and prevent them from running by `brownie test` even
...
+3) Revert the chain changes
+4) Execute the vote
+5) Do (1) and (2) again
+6) Compare the snapshots got during the first and the second scenario runs
+7) The expected outcome is that the voting doesn't change
+
+Current snapshot implementation in kind of MVP and need a number of issues to
+be addressed in the future:
+1) expand the number of storage variables observed
+2) allow modification of the storage variables supposed not to be changed after
```
[the remark this answers is not in the document]
Comment (TheDZhon) [reply README.md:195]: Added

File: tests/common/regression/test_regression_burn_shares.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
+"""
+Tests for lido burnShares method for voting 24/05/2022
```
[the remark this answers is not in the document]
Comment (TheDZhon) [reply tests/common/regression/test_regression_burn_shares.py:2]: fixed

File: tests/common/regression/test_regression_burn_shares.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+start_and_execute_votes = get_start_and_execute_votes_func()
+
+
+@pytest.fixture(scope="module")
+def stranger(accounts):
+    return accounts[0]
+
+
+@pytest.fixture(scope="module")
+def dao_voting_as_eoa(accounts, dao_voting):
```
[the remark this answers is not in the document]
Comment (TheDZhon) [reply tests/common/regression/test_regression_burn_shares.py:19]: Done

File: utils/import_current_vote.py (hunk 3 above)
Comment (arwer13) [reply utils/import_current_vote.py]: Refactored, do you like it?

File: tests/common/regression/test_burn_shares.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    return accounts[0]
+
+
+@pytest.fixture(scope="module", autouse=is_there_any_vote_scripts())
+def autoexecute_vote(dao_voting, helpers, vote_id_from_env):
+    if vote_id_from_env:
+        helpers.execute_vote(
+            vote_id=vote_id_from_env, accounts=accounts, dao_voting=dao_voting, topup='0.5 ether'
+        )
+    elif start_and_execute_votes:
```
Comment (folkyatina) [inline tests/common/regression/test_burn_shares.py:21]: Is this `elif` required here?

Comment (arwer13) [reply tests/common/regression/test_burn_shares.py]: fixed

**Review by TheDZhon — APPROVED** [review]

**Pull request merged**